roxie/ccd/ccdqueue.cpp

```diff
@@ -2114,40 +2115,44 @@ class DelayedPacketQueueChannel : public CInterface
     inline unsigned queryChannel() const { return channel; }
     inline DelayedPacketQueue &queryQueue(unsigned subchannel)
     {
-        if (subchannel > active)
-            active = subchannel;
+        assertex(subchannel);  // Subchannel 0 means primary and is never delayed
+        subchannel -= 1;
+        if (subchannel > maxSeen)
+            maxSeen = subchannel;
         return queues[subchannel];
     }
-    unsigned timeout() const
+    unsigned timeout(unsigned now) const
     {
         unsigned min = (unsigned) -1;
-        for (unsigned queue = 0; queue < active; queue++)
+        for (unsigned queue = 0; queue <= maxSeen; queue++)
         {
-            unsigned t = queues[queue].timeout();
+            unsigned t = queues[queue].timeout(now);
             if (t < min)
                 min = t;
         }
         return min;
     }
     void checkExpired(
+            unsigned now,
 #ifdef ROXIE_SLA_LOGIC
             RoxieQueue &slaQueue,
 #endif
             RoxieQueue &hiQueue, RoxieQueue &loQueue)
     {
-        for (unsigned queue = 0; queue < active; queue++)
+        for (unsigned queue = 0; queue <= maxSeen; queue++)
         {
             queues[queue].checkExpired(
+                    now,
 #ifdef ROXIE_SLA_LOGIC
                     slaQueue,
 #endif
                     hiQueue, loQueue);
         }
     }
 private:
-    DelayedPacketQueue queues[MAX_SUBCHANNEL];
+    DelayedPacketQueue queues[MAX_SUBCHANNEL-1];
```

Comment:
Should have a comment that -1 because it doesn't include when this is the primary

Reply:
added